### PR TITLE
Fix/statistics values

### PR DESCRIPTION
**Pull request by kimberllyflorez**

### CODE REVIEW

**Comment by curregoz** [comment]
@kimberllyflorez Please. Try to keep your commit messages more descriptive. `Advance` isn't a good message for a commit

**Comment by curregoz** [comment]
Please add in the description on the PR what is the objective of this PR

**Comment by kimberllyflorez** [comment]
The object in this PR is fix the value of calorie object in where we got 0.0 now is the correct value of the calculation calorie

File: lib/app/home/provider/home_provider.dart [hunk 1]
```diff
@@ -1,18 +1,42 @@
...
-import 'package:projects_study/data/models/food_items_model.dart';
+import 'package:projects_study/domain/use_cases/profile/calculate_calories_use_case.dart';
+import 'package:rxdart/subjects.dart';
 
 class HomeProvider extends ChangeNotifier {
   final GetMealsUseCase _listMealsUseCase;
+  final CalculateCalorieUseCase _calculateCalorieUseCase;
+  final _state = BehaviorSubject<HomeState>.seeded(
+    const HomeState(
+        calorieObject: '0.0',
```
Comment (curregoz) [inline lib/app/home/provider/home_provider.dart:12]: Its not necessary initialize values of the object HomeState here.

File: lib/app/home/provider/home_provider.dart [hunk 2]
```diff
@@ -1,18 +1,42 @@
...
+  Stream<HomeState> get state => _state;
+
+  HomeProvider(
+    this._listMealsUseCase,
+    this._calculateCalorieUseCase,
+  );
+
+  void init() {
     getListFoodItems();
+    getCalorieObjective();
```
Comment (curregoz) [inline lib/app/home/provider/home_provider.dart:28]: You can join this two calls usin Rx.combine to emit only one time a state

File: lib/app/home/ui/pages/home_page.dart [hunk 3]
```diff
@@ -16,13 +18,14 @@ class HomePage extends StatefulWidget {
...
 
-class _HomePageState extends State<HomePage> {
+class _HomePageState extends PageState<HomePage, HomeProvider> {
   late HomeProvider _homeProvider;
 
   @override
   void initState() {
-    _homeProvider = locator<HomeProvider>();
     super.initState();
+    _homeProvider = locator<HomeProvider>();
```
Comment (curregoz) [inline lib/app/home/ui/pages/home_page.dart:27]: This instace is not necessary.
Use provider object of PageState instead

File: lib/app/search_products/ui/pages/food_tracker_page.dart [hunk 4]
```diff
@@ -20,19 +21,20 @@ class FoodTrackerPage extends StatefulWidget {
...
-  late HomeProvider _foodItemsProvider;
+class _FoodTrackerPageState
+    extends PageState<FoodTrackerPage, FoodTrackerProvider> {
+  late FoodTrackerProvider _foodItemsProvider;
 
   @override
   void initState() {
-    _foodItemsProvider = locator<HomeProvider>();
     super.initState();
+    _foodItemsProvider = locator<FoodTrackerProvider>();
```
Comment (curregoz) [inline lib/app/search_products/ui/pages/food_tracker_page.dart:31]: Same here.  Use provider object of PageState instead

File: lib/app/search_products/provider/food_traker_provider.dart [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+class FoodTrackerProvider extends ChangeNotifier {
+  final GetMealsUseCase _getMealUseCase;
+
+  FoodTrackerProvider(
+    this._getMealUseCase,
+  ) {
+    getListFoodItems();
+  }
+
+  List<FoodItemsModel> meals = [];
```
Comment (curregoz) [inline lib/app/search_products/provider/food_traker_provider.dart:14]: Create a state objetc for this provider

File: lib/domain/use_cases/user_data/level_activity/save_activity_lavel_use_case.dart [hunk 6]
```diff
@@ -9,6 +9,7 @@ class SaveActivityLevelImpl implements SaveActivityLevelUseCase {
   @override
   void call(int level) {
     PreferenceUtils.setBool(PreferenceConst.completedData, true);
+    PreferenceUtils.setInt(PreferenceConst.activity, level);
```
Comment (curregoz) [inline lib/domain/use_cases/user_data/level_activity/save_activity_lavel_use_case.dart:12]: Avoid using the class PreferenceUtils. Use Storage and add this via DI

**Review by curregoz — CHANGES_REQUESTED** [review]

File: lib/domain/use_cases/user_data/level_activity/save_activity_lavel_use_case.dart (hunk 6 above)
Comment (kimberllyflorez) [reply lib/domain/use_cases/user_data/level_activity/save_activity_lavel_use_case.dart:12]: this is solved in another branch

File: lib/app/search_products/provider/food_traker_provider.dart (hunk 5 above)
Comment (kimberllyflorez) [reply lib/app/search_products/provider/food_traker_provider.dart:14]: done

File: lib/app/search_products/ui/pages/food_tracker_page.dart (hunk 4 above)
Comment (kimberllyflorez) [reply lib/app/search_products/ui/pages/food_tracker_page.dart:31]: sorry about that

**Review by curregoz — APPROVED** [review]

**Pull request merged**